src/syrupy/serializers/base.py

```diff
@@ -93,10 +116,21 @@ def post_read(self, index: int = 0) -> None:
     def pre_write(self, data: "SerializableData", index: int = 0) -> None:
         self._ensure_snapshot_dir(index)
 
+    @final
     def write(self, data: "SerializableData", index: int = 0) -> None:
+        """
+        Override _write_snapshot_or_remove_file in subclass to change behaviour
```

Comment:
This method should not be overridden but rather use `get_filepath`, `get_snapshot_name` and the internal `_write_snapshot_or_remove_file` which can be overridden to customise write behaviour

Reply:
"Private" methods aren't part of the public API and so shouldn't be overridden by derived classes. Last time I refactored the abstract class, I explicitly renamed functions to remove the leading underscore if they were overwritten.

Reply:
I had meant to call out that these methods should not be accessed externally on the object instance but can still be overridden. By convention double underscore methods should not be overridden though.

Reply:
https://stackoverflow.com/a/14898239/3072906